inspect/profile.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inspect
+
+import "time"
+
+// Profiler contains a sequence of profiles which are collected over the course of a query execution.
+type Profiler struct {
+	profiles []Profile
```

Comment:
so if we want to test this particular logic (regarding adding profiles, etc), we can do it by mocking out `time.Now()` somehow. there's two ways to do it in go.

1. make `time.Now()` as a member function of `Profiler`,
2. make `now` function package-private, and allow tests to override it.